File: CS/CustomEditor/RepositoryItemFileEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using System.IO;
namespace CustomEditor {
    [UserRepositoryItem("Register")]
    public class RepositoryItemFileEdit : RepositoryItemPopupContainerEdit {
        static RepositoryItemFileEdit() {
            Register();
        }

        public RepositoryItemFileEdit() {
        }


        public override PopupContainerControl PopupControl {
            get {
                if (customPopupControl == null)
                    customPopupControl = new PopupContainerControl();
                return customPopupControl;
            }
            set {

                customPopupControl = value;
            }
        }


        private PopupContainerControl customPopupControl;
        

        public const string FileEditName = "FileEdit";

        public override string EditorTypeName { get { return FileEditName; } }

 
        public override string GetDisplayText(DevExpress.Utils.FormatInfo format, object editValue) {
            return editValue!= null ? Path.GetFileName(editValue.ToString()) : "";
        }

  
        public static void Register() {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(FileEditName, typeof(FileEdit),
                typeof(RepositoryItemFileEdit), typeof(PopupContainerEditViewInfo),
                new ButtonEditPainter(), true, null, typeof(DevExpress.Accessibility.PopupEditAccessible)));
        }
    }
}

File: VB/CustomEditor/RepositoryItemFileEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Drawing
Imports System.IO
Namespace CustomEditor
	<UserRepositoryItem("Register")> _
	Public Class RepositoryItemFileEdit
		Inherits RepositoryItemPopupContainerEdit
		Shared Sub New()
			Register()
		End Sub

		Public Sub New()
		End Sub


		Public Overrides Property PopupControl() As PopupContainerControl
			Get
				If customPopupControl Is Nothing Then
					customPopupControl = New PopupContainerControl()
				End If
				Return customPopupControl
			End Get
			Set(ByVal value As PopupContainerControl)

				customPopupControl = value
			End Set
		End Property


		Private customPopupControl As PopupContainerControl


		Public Const FileEditName As String = "FileEdit"

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return FileEditName
			End Get
		End Property


		Public Overrides Function GetDisplayText(ByVal format As DevExpress.Utils.FormatInfo, ByVal editValue As Object) As String
			Return If(editValue IsNot Nothing, Path.GetFileName(editValue.ToString()), "")
		End Function


		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(FileEditName, GetType(FileEdit), GetType(RepositoryItemFileEdit), GetType(PopupContainerEditViewInfo), New ButtonEditPainter(), True, 0, GetType(DevExpress.Accessibility.PopupEditAccessible)))
		End Sub
	End Class
End Namespace
